server/rest/src/test/java/org/infinispan/rest/resources/WeakSSEListener.java

```diff
@@ -0,0 +1,82 @@
+package org.infinispan.rest.resources;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiFunction;
+
+import org.infinispan.test.TestException;
+import org.infinispan.util.KeyValuePair;
+
+/**
+ * A listener for SSE without ordering restriction.
+ */
+public class WeakSSEListener extends SSEListener {
+   private final ConcurrentMap<String, List<String>> backup = new ConcurrentHashMap<>();
+
+   public void expectEvent(String type, String subString) throws InterruptedException {
+      CompletableFuture<Void> waitEvent = CompletableFuture.supplyAsync(() -> {
+         AtomicBoolean contains = new AtomicBoolean(false);
+         backup.computeIfPresent(type, (k, v) -> {
+            int index = -1;
+            for (int i = 0; i < v.size() && !contains.get(); i++) {
+               if (v.get(i).contains(subString)) {
+                  contains.set(true);
+                  index = i;
+               }
+            }
+
+            if (index >= 0) v.remove(index);
+            return v;
+         });
+
+         while (!contains.get()) {
+            try {
+               KeyValuePair<String, String> event = events.poll(10, TimeUnit.SECONDS);
+               assertNotNull(event);
+
+               if (type.equals(event.getKey())) {
+                  if (event.getValue().contains(subString)) {
+                     contains.set(true);
+                     break;
+                  } else {
+                     backup.compute(event.getKey(), compute(event));
+                  }
+               } else {
+                  backup.compute(event.getKey(), compute(event));
+               }
+            } catch (InterruptedException e) {
```

Comment:
This can be simplified to:

```java
               if (type.equals(event.getKey()) && event.getValue().contains(subString)) {
                  contains.set(true);
                  break;
               } else {
                  backup.compute(event.getKey(), compute(event));
               }
```

We can then just define the compute `BiFunction` inline as it's only required once.